src/transforms/eigenanalysis.jl

```diff
@@ -69,32 +78,35 @@ function apply(transform::EigenAnalysis, table)
   # project the data
   Z = Y * S
 
+  # discarted and selected coluns
+  D = Z[:, ndim+1:end]
```

Comment:
I think we don't need to introduce a new matrix in the cache. We can simply post-process the matrix S and S^-1 to discard columns right?

Reply:
You're totally right. I will change it.